pyccel/ast/numpyext.py

```diff
@@ -1160,7 +1067,7 @@ def __new__(cls, a, dtype=None, order='K', subok=True):
 
 #=======================================================================================
 
-class Bounds(Basic):
+class Bounds(Basic, PyccelAstNode):
```

Comment:
I don't think Numpy has a `bounds` function. Is this class ever used?

Reply:
I'm not sure maybe it's used

Reply:
A grep search shows that this class has an associated `_print_Bounds` function in `pyccel/codegen/printing/fcode.py` however it is not mentioned anywhere else in the codebase. The codacy-coverage results also indicate that it is not used.

The printing function looks like this:
```
def _print_Bounds(self, expr):
    var = expr.var
    rank = var.rank
    bounds = []
    for i in range(0, rank):
        l = 'lbound({var},{i})'.format(var=self._print(var),
                                       i=self._print(i+1))
        u = 'ubound({var},{i})'.format(var=self._print(var),
                                       i=self._print(i+1))

        bounds.append('{lbound}:{ubound}'.format(lbound=l, ubound=u))
    bounds = ','.join(bounds)
    return bounds
```
which means that Bounds does not represent any python concept

Reply:
I think we used that before to allocate memory for unknown shapes like this example
```
def f(x):
     y = zeros_like(x)
```
because here the shape of x is unknown

Reply:
So I suppose we now handle this with classes `Shape` and `PyccelArraySize` in `numpyext`? The array bounds are never needed, because in the generated Fortran code we use 0-based indexing. The same goes for the generated C code.

Reply:
yes you are right

Reply:
@saidctb If you agree, please remove class `Bounds` from module `numpyext` and `_print_Bounds` from `fcode`. Also check if we have a `_visit_Bounds` somewhere else.

Reply:
okay